File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mini=prices[0]
        max_profit=0
        for i in range(1,len(prices)):
            if prices[i]>prices[i-1]:
                max_profit+=(prices[i]-prices[i-1])
                mini=prices[i]
            if prices[i]<mini:
                mini=prices[i]
        return max_profit

File: 122-best-time-to-buy-and-sell-stock-ii/122-best-time-to-buy-and-sell-stock-ii.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini=prices[0];
        int max_profit=0;
        for (int i=1;i<prices.size();i++){
            if (prices[i]>prices[i-1]){
                max_profit+=prices[i]-mini;
                mini=prices[i];
            }
            if (prices[i]<mini)
                mini=prices[i];
        }
        return max_profit;
    }
};